deepchem/rl/a3c.py

```diff
@@ -263,7 +263,7 @@ def create_rollout(self):
     for i in range(self.a3c.max_rollout_length):
       if self.env.terminated:
         break
-      state = self.env.state
+      state = copy.deepcopy(self.env.state)
```

Comment:
While I understand that decision -- I don't think it will be clear to most users and will lead to users mutating state in the Environment with step() and reset().  We should at least document it on the Environment object, state property, and step function.

Doing it here is safer and easier to reason about -- but will lead to worse performance in the case of someone actually making their state immutable, and creating a new one each time.